packages/devtools_app/lib/src/flutter/status_line.dart

```diff
@@ -2,26 +2,225 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:async';
+
 import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:url_launcher/url_launcher.dart' as url_launcher;
+import 'package:vm_service/vm_service.dart';
 
 import '../../devtools.dart' as devtools;
+import '../globals.dart';
+import '../service_manager.dart';
+import 'common_widgets.dart';
+import 'notifications.dart';
+import 'screen.dart';
+
+const statusLineHeight = 24.0;
 
 /// The status line widget displayed at the bottom of DevTools.
+///
+/// This displays information global to the application, as well as gives pages
+/// a mechanism to display page-specific information.
 class StatusLine extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
-    // TODO(devoncarew): Break this into an isolates area, a connection status
-    // area, a page specific area, and a documentation link area.
-
     final textTheme = Theme.of(context).textTheme;
 
+    final Screen currentScreen = Provider.of<Screen>(context);
+
+    final List<Widget> children = [];
+
+    // Have an area for page specific help (always docked to the left).
+    children.addAll([
+      Expanded(
+        child: Align(
+          alignment: Alignment.centerLeft,
+          child: buildHelpUrlStatus(context, currentScreen, textTheme),
+        ),
+      ),
+      BulletSpacer(),
+    ]);
+
+    // Optionally display page specific status.
+    if (currentScreen != null && currentScreen.providesStatus) {
```

Comment:
nit: can we instead omit the status if buildPageStatus returns null?
This UI would look odd if a user ever neglected to keep providesStatus in sync with buildPageStatus